project_name/features/embedding_processing.py

```diff
@@ -0,0 +1,56 @@
+import pandas as pd
+from imblearn.under_sampling import RandomUnderSampler
+from sklearn.decomposition import PCA
+
+
+def balance_dataframe(df):
+    """
+    Balance a dataframe by undersampling the majority class
+
+    Args:
+        df (dataframe): The dataframe to balance
+
+    Returns:
+        dataframe: The balanced dataframe
+    """
+
+    # Separate features and labels
+    features = df.drop(columns=["label"])
+    labels = df["label"]
+
+    # Balance dataset
+    rus = RandomUnderSampler()
+    features_resampled, labels_resampled = rus.fit_resample(features, labels)
+
+    # Combine features and labels
+    balanced_dataframe = pd.DataFrame(features_resampled)
+    balanced_dataframe["label"] = labels_resampled
+
+    return balanced_dataframe
+
+
+def reduce_dimensions(df, n_dimensions):
```

Comment:
i'd add PCA in the name, or maybe as a parameter, so that later we could switch from one method to another with ease.

I'd say adding PCA to the name would be better